Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -1123,5 +1130,79 @@ private void PrintNetworkInformation(NetworkConfiguration config, bool debug = t
+        ///
+        /// If multiple addresses are available:
+        /// - prefer Unique Local Addresses (ULA) or Site Local Addresses over other (global) addresses.
+        /// - all things equal, prefer IPv6 over IPv4 as the former is more state-of-the-art.
+        /// </summary>
+        private void InitializeServerLanAddress(NetworkConfiguration config)
+        {
+            lock (_initLock)
+            {
+                int rank = 0;   // ranks most preferable ip address
+                IPAddress ipSrv = IPAddress.None;
+                NetworkInterface[] adapters = NetworkInterface.GetAllNetworkInterfaces();
+
+                foreach (var adapter in adapters)
+                {
+                    if (adapter.NetworkInterfaceType == NetworkInterfaceType.Loopback)
+                    {
+                        continue;
+                    }
+
+                    // check if the network interface serves as IPv4 gateway
+                    bool haveIP4GateWay = false;
+                    foreach (var address in adapter.GetIPProperties().GatewayAddresses)
+                    {
+                        haveIP4GateWay |= address.Address.AddressFamily == AddressFamily.InterNetwork;
+                    }
+
+                    // if the network interface is used as gateway, and has an acceptable IP address for this server,
+                    // then this address is assumed to be accessible by other devices on the LAN.
+                    foreach (var inf in _interfaces)
+                    {
+                        IPAddress ip = inf.Address;
+
+                        if (inf.Name.Equals(adapter.Name, StringComparison.OrdinalIgnoreCase) &&
+                            !ip.Equals(IPAddress.Loopback) && !ip.Equals(IPAddress.IPv6Loopback))
+                        {
+                            if (ip.AddressFamily == AddressFamily.InterNetworkV6)
+                            {
+                                if (IsIPv6Enabled)
+                                {
+                                    if (rank < 3 && (ip.IsIPv6SiteLocal || ip.IsIPv6UniqueLocal))
+                                    {
+                                        rank = 3;
+                                        ipSrv = ip;
+                                    }
+                                    else if (rank < 1)
+                                    {
+                                        rank = 1;
+                                        ipSrv = ip;
+                                    }
+                                }
+                            }
+                            else if (haveIP4GateWay && IsIPv4Enabled && rank < 2)
+                            {
+                                rank = 2;
+                                ipSrv = ip;
+                            }
+                        }
+                    }
+
+                    ServerLanAddress = ipSrv.ToString();
```

Comment:
The server can listen on multiple NICs. Assigning a single IP as _the_ LAN address will work for most setups but might not work correctly when there's multiple LAN subnets involved. Shouldn't we take the caller IP into account?

Reply:
Yes that is a tough one.

My main goal was to add functionality so that Chromecast works out of the box for simple systems. With this PR one can for instance just download https://repo.jellyfin.org/releases/server/windows/stable/installer/jellyfin_xxx_windows-x64.exe and have Jellyfin server and client running with Chromecast on one single PC without any additional configuration (apart from the media folders of course.) From my own experience this helps a lot in getting up to speed with Jellyfin for first time/non-expert users.

For more extended systems like multiple NIC's it might be possible in some cases to autoconnect chromecast receivers without additional configuration, but the solution probably will be more complicated: I guess that after the Jellyfin webclient connects (using the Cast button) to a cast device, it should use the friendlyName returned by the onSessionConnected event to identify the device. The IP address (subnet) of the cast device has to be found by mDNS discovery, which a javascript in a browser is not able to execute. Jellyfin server can do so, however, being a standard application, so Jellyfin-web should probably submit the friendlyName to the server in an API call with a request to resolve its IP address (or actually: the serveraddres on its subnet).

Apart from the coding complexity I am not sure if this approach works in all cases without configuration, and more advanced systems might require explicit configuration anyhow, including assigning of chromecast devices.

So, if there is a good use-case for autoconnect to chromecast I could investigate this in a future PR.

Note however that this PR does add autoconnect for "simple" system, and does *not* make the situation worse for more advanced systems: in the latter case the autoconnect might not work *with* the PR, but neither did if *without* the PR ;-)

I would be quite interested in your ideas and suggestions on this!